fhir-term-graph/src/main/java/com/ibm/fhir/term/graph/impl/FHIRTermGraphImpl.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.term.graph.impl;
+
+import static com.ibm.fhir.term.graph.util.FHIRTermGraphUtil.setRootLoggerLevel;
+import static java.util.Objects.requireNonNull;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.janusgraph.core.JanusGraph;
+import org.janusgraph.core.JanusGraphFactory;
+import org.janusgraph.core.JanusGraphIndexQuery.Result;
+import org.janusgraph.core.JanusGraphVertex;
+import org.janusgraph.core.PropertyKey;
+import org.janusgraph.core.RelationType;
+import org.janusgraph.core.schema.JanusGraphManagement;
+import org.janusgraph.core.schema.Mapping;
+
+import com.ibm.fhir.term.graph.FHIRTermGraph;
+
+public class FHIRTermGraphImpl implements FHIRTermGraph {
+    private static final Logger log = Logger.getLogger(FHIRTermGraphImpl.class.getName());
+
+    private static final String STORAGE_READ_ONLY = "storage.read-only";
+
+    private JanusGraph graph;
+    private GraphTraversalSource traversal;
+
+    public FHIRTermGraphImpl(Configuration configuration) {
+        requireNonNull(configuration);
+        try {
+            graph = open(configuration);
+            traversal = graph.traversal();
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    private JanusGraph open(Configuration configuration) {
+        log.info("Opening graph...");
+
+        boolean readOnly = configuration.getBoolean(STORAGE_READ_ONLY, false);
+        configuration.setProperty(STORAGE_READ_ONLY, false);
+
+        setRootLoggerLevel(ch.qos.logback.classic.Level.INFO);
```

Comment:
interesting.... should we be using  logback across the system? or is this just for Janus?

Reply:
This is just for JanusGraph. The default logging level is DEBUG which generates way too many messages.